python/ray/experimental/data/dataset.py

```diff
@@ -805,8 +805,31 @@ def iter_batches(self,
         Returns:
             A list of iterators over record batches.
         """
-
-        raise NotImplementedError  # P1
+        blocks = list(iter(self._blocks))
+        for curr_block_idx in range(len(blocks)):
+            # Prefetch blocks.
+            ray.wait(
```

Comment:
Nice!

Reply:
We should also port iter_rows() to use iter_batches() under the hood--- this should automatically enable prefetch IIUC since iter_batches() does.

Also, if we did that then you don't need to add the extra unit test, since then the existing test covers iter batches as well (take -> iter_rows -> iter_batches).

Reply:
Hmm maybe this doesn't make sense unless we support batch format == None/"block"